### PR TITLE
Build tools guide v2

### LINKED ISSUE #5151: The Build Tools guide is outdated/deprecated
Re: https://pantheon.io/docs/guides/build-tools

Priority: Medium

## Issue Description
The Build Tools guide is outdated and should be updated for version 2 of the package.

### PR DESCRIPTION
**Pull request by ataylorme**
Closes #5151

**Work in Progress, do NOT merge**

## Effect
PR includes the following changes:
- Update the guide at `/guides/build-tools` from Build Tools version 1 to version 2

## Remaining Work
- [ ] Rework all guide sections
  - [x] What is Build Tools?
    - [x] Project purpose
    - [x] Build Tools workflow overview
    - [x] When to use Build Tools
    - [x] Build Tools components/related repositories
  - [x] Installing Build Tools
  - [x] Interactive `build:project:create` command generator
  - [ ] Example repo/Continuous Integration breakdown
    - This should be in an appendix/additional information section
    - [ ] Differences between providers (GitHub/CircleCI, GitLab, BitBucket)
    - [ ] Differences between CMS (WordPress/Drupal 8)
  - [ ] Drupal 8/WordPress walkthroughs
    - [ ] Install a new package with Composer
    - [ ] PR to multidev
    - [ ] Config change
    - [ ] Export/commit config (Quicksilver pushback)
      - For WordPress, [wp-cfm](https://pantheon.io/blog/webops-just-got-better-pantheon-wp-cfm-integration) can be used
  - [ ] Recap
    - Encourage users to customize the example repository. Don't need Behat? Remove it. Need a NodeJS build step? Add it.
- [ ] Technical review
- [ ] Copy Review

## Post Launch
**Do not remove** - To be completed by the docs team upon merge:
- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] Update Status Report
- [ ] Archive from **Done** in [Main Project](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by ataylorme** [comment]
@alexfornuto I think I have persistent state working in the Build Tools Guide! Check out the first few pages of https://5151build-static-docs.pantheonsite.io/docs/guides/build-tools

I'm using local storage to save the value. Now I just need to rewrite the entire guide, customizing for all the CMS/CI combos 😅

File: source/content/guides/build-tools/01-introduction.md [hunk 1]
```diff
@@ -20,104 +20,77 @@ image: buildToolsGuide-thumb
...
+## Build Tools Workflow
+Build Tools projects extend the [Pantheon workflow](https://pantheon.io/docs/pantheon-workflow) by adding an external Git provider and a CI service.
 
-    ```bash
-    composer create-project -n --no-dev -d $HOME/.terminus/plugins pantheon-systems/terminus-composer-plugin:~1
-    ```
+In this workflow, only files unique to the project are tracked as part of external Git repository. The CI service then builds a production artifact, deploys the fully-build site to Pantheon, and runs a suite of automated tests.
 
-7. Install the [Terminus Drupal Console Plugin](https://github.com/pantheon-systems/terminus-drupal-console-plugin):
+A diagram visualizing this workflow with GitHub and CircleCI is below but the concepts are the same when working with other providers.
```
Comment (alexfornuto) [inline source/content/guides/build-tools/01-introduction.md]: @ataylorme would you like is to recreate that image with the tools abstracted?

Comment (ataylorme) [reply source/content/guides/build-tools/01-introduction.md]: @alexfornuto my plan is to get through rewriting the guide and making it agnostic, then to ask design for help with images

**Comment by ataylorme** [comment]
WordPress variations of the guide should probably use [WP-CFM](https://pantheon.io/blog/webops-just-got-better-pantheon-wp-cfm-integration)

**Comment by sugaroverflow** [comment]
Closing in favour of separating into #5444 the docs and #5446 the gatsby component PR.

**Pull request closed without merging**